src/Microsoft.AspNet.Mvc.Razor/Resources.resx

```diff
@@ -195,4 +195,9 @@
   <data name="RazorPage_InvalidTagHelperIndexerAssignment" xml:space="preserve">
     <value>Unable to perform '{0}' assignment. Tag helper property '{1}.{2}' must not be null.</value>
   </data>
+  <data name="CouldNotResolveApplicationRelativeUrl_TagHelper" xml:space="preserve">
+    <value>Unexpected return value from '{1}.{2}' for URL '{0}'. If you have overridden the '{1}' service, change '{2}' to replace only the '~/' prefix. Otherwise, add the following directive to your Razor page to disable application-relative URL resolution:
```

Comment:
Error messages shouldn't assign blame. Remove all the "you" stuff and make it objective: `"If the blah service has been overridden... blah blah. Add the Foo directive to the Razor page...."`